From: Colin Watson <cjwatson@ubuntu.com>
Date: Thu, 1 Feb 2018 13:46:47 +0000
Subject: skip tests if an upstream HTTP proxy is set

shadowsocks-libev doesn't support connecting through an upstream HTTP
proxy, so if the test environment requires that then there's nothing we can
do.  (In future it might be possible to set up a local server to avoid this
problem.)
---
 tests/test.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/tests/test.sh b/tests/test.sh
index 1dcd15f..b666eb1 100755
--- a/tests/test.sh
+++ b/tests/test.sh
@@ -27,6 +27,11 @@ function run_test {
 [ -d src -a -x src/ss-local ] &&
     BIN="--bin src/"
 
+if [ "$http_proxy" ]; then
+    echo "SKIP: shadowsocks-libev does not support an upstream HTTP proxy"
+    exit 0
+fi
+
 run_test python tests/test.py $BIN -c tests/aes.json
 run_test python tests/test.py $BIN -c tests/aes-gcm.json
 run_test python tests/test.py $BIN -c tests/aes-ctr.json
